src/server/api/routers/upload.ts

```diff
@@ -0,0 +1,78 @@
+import { TRPCError } from "@trpc/server";
+import { string, z } from "zod";
+import {
+  createTRPCRouter,
+  protectedProcedure,
+  publicProcedure,
+} from "~/server/api/trpc";
+
+export const uploadRouter = createTRPCRouter({
+  getFiles: protectedProcedure.mutation(async ({ ctx, input }) => {
```

Comment:
This is a `query`, not a `mutation`.
[Read more](https://github.com/trpc/trpc/discussions/1638#discussioncomment-2341051)